core/src/main/java/jenkins/security/LastGrantedAuthoritiesProperty.java

```diff
@@ -48,14 +48,20 @@ public UserProperty reconfigure(StaplerRequest req, JSONObject form) throws Form
     public GrantedAuthority[] getAuthorities() {
         String[] roles = this.roles;    // capture to a variable for immutability
 
-        GrantedAuthority[] r = new GrantedAuthority[roles==null ? 1 : roles.length+1];
-        r[0] = SecurityRealm.AUTHENTICATED_AUTHORITY;
-        if (roles != null) {
-            for (int i = 1; i < r.length; i++) {
-                r[i] = new GrantedAuthorityImpl(roles[i - 1]);
+        if(roles == null){
+            return new GrantedAuthority[]{SecurityRealm.AUTHENTICATED_AUTHORITY};
+        }
+
+        String authenticatedRole = SecurityRealm.AUTHENTICATED_AUTHORITY.getAuthority();
+        List<GrantedAuthority> grantedAuthorities = new ArrayList<>(roles.length);
+        for (int i = 0; i < roles.length; i++){
+            // to avoid having twice that role
+            if(!authenticatedRole.equals(roles[i])){
+                grantedAuthorities.add(new GrantedAuthorityImpl(roles[i]));
```

Comment:
If `this.roles` is not null, but doesn't contain `SecurityRealm.AUTHENTICATED_AUTHORITY`, then `SecurityRealm.AUTHENTICATED_AUTHORITY` won't be added to the result. Unless I'm missing something I think you want to add `AUTHENTICATED_AUTHORITY` to the the list before the loop to make sure it's included, right? (And initialize the list to `roles.length + 1`).